src/execution/as_execution.rs

```diff
@@ -24,6 +24,15 @@ impl MassaModule for ASModule {
         &self.bytecode
     }
     fn execution(&self, instance: &Instance, function: &str, param: &str) -> Result<Response> {
+        // sub initial metering cost
+        let metering_initial_cost = settings::metering_initial_cost();
+        let remaining_gas = get_remaining_points(&self.env)?;
+        if metering_initial_cost > remaining_gas {
+            bail!("Not enough gas for initial cost")
+        }
+        set_remaining_points(&self.env, remaining_gas - metering_initial_cost)?;
```

Comment:
can you check that this works when called before calling wasm_func.call ?

Reply:
(and that calling call() doesn't reset the gas :) )

Reply:
A unit test for this feature was written and all tests run without error in this branch. Is that enough?

Reply:
yes :)